### PR TITLE
docs(breaking): update stencil v3.0.0 breaking changes

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Our `BREAKING_CHANGES.md` file has grown out of date
GitHub Issue Number: N/A

## What is the new behavior?

this commit updates the `BREAKING_CHANGES.md` document to include all
breaking changes made to date in the stencil v3.0.0-dev branch

this commit updates `BREAKING_CHANGES.md` by adding a table of contents
to the document, as well as an executive summary. the goal of this
commit is to align with our sibling project, the Ionic Framework, it
terms of layout of the document.

keeping with that philosophy, existing header tags have been updated to
be one level 'smaller' to accommodate the new structure

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

Links work as expected

### CODE REVIEW

File: BREAKING_CHANGES.md [hunk 1]
```diff
@@ -1,38 +1,303 @@
...
+     }
+ }
++export interface MyComponentCustomEvent<T> extends CustomEvent<T> {
++    detail: T;
++    target: HTMLMyComponentElement;
++}
+ declare global {
+     interface HTMLMyComponentElement extends Components.MyComponent, HTMLStencilElement {
+}
+```
```
Comment (tanner-reits) [inline BREAKING_CHANGES.md:148]: The types were already getting imported, the only change is that they get re-exported now.

```suggestion
```diff
 import { HTMLStencilElement, JSXBase } from "@stencil/core/internal";
 import { NameType } from "./components/my-component/my-component";
+export { NameType } from "./components/my-component/my-component";
 export namespace Components {
     interface MyComponent {
        "first": NameType;
     }
 }
 export interface MyComponentCustomEvent<T> extends CustomEvent<T> {
     detail: T;
     target: HTMLMyComponentElement;
 }
 declare global {
     interface HTMLMyComponentElement extends Components.MyComponent, HTMLStencilElement {
}
```
```

Comment (rwaskiewicz) [reply BREAKING_CHANGES.md:148]: @tanner-reits can you do me a favor and transfer this over to https://github.com/ionic-team/stencil-site/pull/922?

This PR is still a draft/isn't ready to be reviewed yet - I wanted to use the stencil-site version as the source of truth for review comments (then once that PR is good, I'll transfer things over to this one)

**Comment by rwaskiewicz** [comment]
@alicewriteswrongs && @tanner-reits  this should be nearly identical to https://github.com/ionic-team/stencil-site/pull/922, with a few exceptions:
- the heading levels are different since this document is more nested than the website
- links are different (linking in the site is different than the markdown file)
- minor wording differences (how we reference "this page" on the site vs "a page on the site" from the MD file)

**Review by alicewriteswrongs — APPROVED** [review]
looks good to me!

**Review by tanner-reits — APPROVED** [review]

**Comment by rwaskiewicz** [comment]
@dependabot rebase

**Pull request merged**